### DISCUSSION TITLE
Install it to the system's root drive (without data loss)? I'd like to add it to my main UEFI to be able to use it as a recovery boot

### DISCUSSION THREAD

**Discussion by levihb** [ventoy/Ventoy#1172] [General]
I've put this as general, as I'm not sure if there is a way to do it, if there isn't it's more of an idea.

I love Ventoy, but I'd really like a way to be able to install it to my main system's SSD. I would like to be able to add it to my EFI partition, and then add an exfat partition in order to load the images from there. I'd like this for the following reasons:

The main one is for recovery. Being able to just boot into an image of my choice when there's an issue with the main OS would be very useful. I don't always have direct access to a USB drive. Also of course easy to reinstall an OS.

Being able to maintain a list of images. I generally keep a bunch of system images in `~/images`. I'd love to be able to just mount that from the Ventoy partition, so any images I place in there just automatically show up on booting to it (I would also like to make a simple script to e.g. automatically download the latest Arch image weekly).

Being able to simply test other operating systems. I sometimes boot into e.g. an Ubuntu live CD in order to test something. This would make it much easier and faster, without the limitations of a VM.

I have tried doing some basic things. E.g. just mounting my Ventoy USB drive, and copying over the EFI partition files to my EFI partition. It shows up in my bios, but I just get placed into a grub command line? I'm guessing it's related to the UUID of the exfat partition being incorrect, or something else?

I think this would be super beneficial. Maybe harder to implement, so how about you allow us generate the EFI files directly, specifying which partitions we will use?

**Comment by bforcier299** [comment]
Hello,

Has this been investigated?

Thanks,

**Reply by levihb** [reply]
I gave up and just permanently left a Ventoy SD card in my laptop. Works well for me since I never use SD cards. Would only work on laptops where the SD card is a click in one that's completely recessed - not those cheap ones where only the front half of the SD card goes in and the rest sticks out.

Also I bought one of those NVMe to 10Gbit USB-C adapters on my keychain, threw an old 500GB Samsung SSD in it, Ventoy'ed that, and use it for Ventoy as well as file storage (Ventoy doesn't care if you load up the iso partition with other files). I love this thing because it's blazingly fast, and I can put tons of images on there, so I use that for my desktop. It's fun installing W10, since the "copying files" part is now like "0... 100" in just a few seconds - feels so much better after years of slow USB 2 drives, then all those USB 3 drives which always seemed to run at barely above USB 2 speeds. Has totally replaced all my USB drives.

**Reply by bforcier299** [reply]
yeah, still USB2 to start install and forget is quite nice...

Thinking about getting a TB4/USB 3.2 NVME drive that would host Ventoy and appropriate install and recovery clonezilla images as well as the trifecta ISOs: clonezilla, shredos & gparted plus any installer I'd need. 

Thanks for your follow up!
